### ISSUE TITLE
makeotf: Within a ‘vkrn’ feature, value record format A should set YAdvance

### ISSUE THREAD

**Issue by brawer** [adobe-type-tools/afdko#85]
The OpenType Feature File Specification says in [section 2.e.iv](http://www.adobe.com/devnet/opentype/afdko/topic_feature_file_syntax.html#2.e.iv) about Value Record Format A: “Here the ‹metric› represents an X advance adjustment, except when used in the ‘vkrn’ feature, in which case it represents a Y advance adjustment.”

However, with makeotf v2.0.90 of Nov 19 2015, the following feature file:
```
languagesystem DFLT dflt;
feature vkrn {
    position A -100;
} vkrn;
```

produces the following GPOS table when dumped with ttx:
```
<?xml version="1.0" encoding="UTF-8"?>
<ttFont sfntVersion="true" ttLibVersion="3.0">

  <GPOS>
    <Version value="1.0"/>
    <ScriptList>
      <!-- ScriptCount=1 -->
      <ScriptRecord index="0">
        <ScriptTag value="DFLT"/>
        <Script>
          <DefaultLangSys>
            <ReqFeatureIndex value="65535"/>
            <!-- FeatureCount=1 -->
            <FeatureIndex index="0" value="0"/>
          </DefaultLangSys>
          <!-- LangSysCount=0 -->
        </Script>
      </ScriptRecord>
    </ScriptList>
    <FeatureList>
      <!-- FeatureCount=1 -->
      <FeatureRecord index="0">
        <FeatureTag value="vkrn"/>
        <Feature>
          <!-- LookupCount=1 -->
          <LookupListIndex index="0" value="0"/>
        </Feature>
      </FeatureRecord>
    </FeatureList>
    <LookupList>
      <!-- LookupCount=1 -->
      <Lookup index="0">
        <!-- LookupType=1 -->
        <LookupFlag value="0"/>
        <!-- SubTableCount=1 -->
        <SinglePos index="0" Format="1">
          <Coverage>
            <Glyph value="A"/>
          </Coverage>
          <ValueFormat value="4"/>
          <Value XAdvance="-100"/>
        </SinglePos>
      </Lookup>
    </LookupList>
  </GPOS>

</ttFont>
```

Instead of `<Value XAdvance="-100"/>`, shouldn’t this be `<Value YAdvance="-100"/>` since the value record format A appears inside a `vkrn` feature?

**Comment by readroberts** [comment]
This is indeed a bug. The feature specification says explicitly "Here the <metric> represents anÃ—advance adjustment, except when used in the 'vkrn' feature, in which case it represents a Y advance adjustment. “.  For the moment, you can work around it by using the record value format B: < 0 0 0 -100>.

**Comment by readroberts** [comment]
As Ken Lunde pointed out to me, the logic does work correctly for a kern pair. The bug exists only for a single glyph positioning adjustment. I will fix this, but out of curiosity, do you have a use case for a single glyph positioning statement in vkrn?

**Comment by kenlunde** [comment]
The '[vpal](https://www.microsoft.com/typography/otspec/features_uz.htm#vpal)' GSUB feature is probably the better vehicle for such a metrics adjustment.

**Comment by brawer** [comment]
Should “vpal” (and other vertical features) trigger the vertical interpretation of ValueRecord format A? From the current spec, the answer seems “no” — should the spec could be adjusted?

**Comment by readroberts** [comment]
Seems like a good idea to me.

**Comment by readroberts** [comment]
Fixed, in makeotf and in feature file spec v 1.19
